### ISSUE TITLE
Overriding `abstractmethod` should not trigger `missing-override-decorator`

### ISSUE THREAD

**Issue by cmp0xff** [astral-sh/ty#4307]
### Summary

https://play.ty.dev/e4e7566f-a444-4c2c-9cb7-db3833bcd1dd

```py
from abc import ABC, abstractmethod
class A(ABC):
    @abstractmethod
    def f(self) -> None:
        pass

class B(A):
    def f(self) -> None:  # ty should not complain `missing-override-decorator`
        pass
```

### Version

playground

**Comment by MichaReiser** [comment]
I think what ty does here is intentional and correct. The idea here is that, if you remove `A.f`, the `@override` decorator will now warn you that it no longer overrides a inherited method.

**Comment by cmp0xff** [comment]
Thank you @MichaReiser , it makes sense.
